### PR TITLE
[Tikyi] iP

### PR DESCRIPTION
**Pull request by tikimonarch**

### CODE REVIEW

File: src/main/java/Duke.java [hunk 1]
```diff
@@ -1,10 +1,69 @@
...
+            if (command.equals("bye")) {
+                break;
+            } else if (command.equals("list")) {
+                if (size == 0) {
+                    System.out.print(border + "Empty\n" + border);
+                } else {
+                    System.out.print(border + "Here are the tasks in your list:\n");
+                    for (int i = 0; i < size; i++) {
+                        System.out.println(" " + Integer.toString(i + 1) + "." + list[i]);
+                    }
```
Comment (1-Karthigeyan-1) [inline src/main/java/Duke.java]: Please avoid deep nesting

File: src/main/java/Duke.java [hunk 2]
```diff
@@ -1,10 +1,69 @@
...
+                list[Integer.valueOf(arrOfStr[1])-1] = temp;
+                System.out.print(border + " Nice! I've marked this task as done:\n" );
+                System.out.println("  " + list[Integer.valueOf(arrOfStr[1])-1]);
+                System.out.print(border);
+                continue;
+            } else {
+                if (command.contains("todo")) {
+                    tag = "[T]";
+                    command = command.replaceAll("todo ", "");
+                    list[size] = tag + "[✗] " + command;
```
Comment (1-Karthigeyan-1) [inline src/main/java/Duke.java]: I think it is possible to group similar code fragments together via refactoring. I have noticed a few similar areas along with the code.

File: src/main/java/Duke.java [hunk 3]
```diff
@@ -1,10 +1,69 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
 public class Duke {
-    public static void main(String[] args) {
+    public static void main(String[] args) throws IOException {
```
Comment (1-Karthigeyan-1) [inline src/main/java/Duke.java]: is it possible to shorten the main method?

File: src/main/java/Duke.java [hunk 4]
```diff
@@ -1,10 +1,69 @@
...
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        String border = "____________________________________________________________\n";
+        System.out.print(border + " Hello! I'm Duke\n" + " What can I do for you?\n" + border);
+        String command;
+        int size = 0;
+        String temp;
+        String tag = "[ ]";
+        String[] temp2 = new String[2];
+        String[] list = new String[100];
```
Comment (1-Karthigeyan-1) [inline src/main/java/Duke.java]: is it possible to avoid magic numbers in the array size?

**Review by 1-Karthigeyan-1 — COMMENTED** [review]
Overall, I think the method can be shortened for better readability

File: src/main/java/Duke.java (hunk 4 above)
Comment (zhixiangteoh) [inline src/main/java/Duke.java]: Perhaps declare a final variable for the maximum number of tasks, as follows:
`static final int MAX_LIST_COUNT = 100` ?

File: src/main/java/Duke.java [hunk 5]
```diff
@@ -1,10 +1,69 @@
...
+                    System.out.print(border + "Empty\n" + border);
+                } else {
+                    System.out.print(border + "Here are the tasks in your list:\n");
+                    for (int i = 0; i < size; i++) {
+                        System.out.println(" " + Integer.toString(i + 1) + "." + list[i]);
+                    }
+                    System.out.print(border);
+                }
+                continue;
+            } else if (command.contains("done")) {
```
Comment (Zhangyilin0203) [inline src/main/java/Duke.java]: It will be better to create another method called donetask and put your code inside the method

File: src/main/java/Duke.java [hunk 6]
```diff
@@ -1,10 +1,69 @@
...
+                String[] arrOfStr = command.split(" ", 2);
+                temp = list[Integer.valueOf(arrOfStr[1])-1].replace('✗','✓');
+                list[Integer.valueOf(arrOfStr[1])-1] = temp;
+                System.out.print(border + " Nice! I've marked this task as done:\n" );
+                System.out.println("  " + list[Integer.valueOf(arrOfStr[1])-1]);
+                System.out.print(border);
+                continue;
+            } else {
+                if (command.contains("todo")) {
```
Comment (Zhangyilin0203) [inline src/main/java/Duke.java]: it will be better to create a todo class and put the behavior of todo inside the class

File: src/main/java/Duke.java [hunk 7]
```diff
@@ -1,10 +1,69 @@
...
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        String border = "____________________________________________________________\n";
+        System.out.print(border + " Hello! I'm Duke\n" + " What can I do for you?\n" + border);
+        String command;
+        int size = 0;
+        String temp;
+        String tag = "[ ]";
+        String[] temp2 = new String[2];
```
Comment (Zhangyilin0203) [inline src/main/java/Duke.java]: could you let the use of temp2 be clearer by changing its name?

File: src/main/java/Duke.java [hunk 8]
```diff
@@ -1,10 +1,69 @@
...
 public class Duke {
-    public static void main(String[] args) {
+    public static void main(String[] args) throws IOException {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        String border = "____________________________________________________________\n";
```
Comment (zhixiangteoh) [inline src/main/java/Duke.java]: Maybe extract this as a constant, like so:

`public static final String BORDER = "____________________________________________________________\n"`?

File: src/main/java/Duke.java [hunk 9]
```diff
@@ -1,10 +1,69 @@
...
+        System.out.print(border + " Hello! I'm Duke\n" + " What can I do for you?\n" + border);
```
Comment (zhixiangteoh) [inline src/main/java/Duke.java]: Perhaps extract this as a greeting method or constant to better align with SLAP paradigm?

File: src/main/java/Duke.java [hunk 10]
```diff
@@ -1,10 +1,69 @@
...
+        String command;
```
Comment (zhixiangteoh) [inline src/main/java/Duke.java]: Maybe this need not be declared here, since you don't use it outside of the main while loop?

Comment (zhixiangteoh) [reply src/main/java/Duke.java]: Perhaps "command" is not the best name for this variable, since in your main while loop command is assigned to the whole line from `read.readLine()`; would it be better to name it something like `userInput` or `userInputLine` instead?

File: src/main/java/Duke.java [hunk 11]
```diff
@@ -1,10 +1,69 @@
...
+                    }
+                    System.out.print(border);
+                }
+                continue;
+            } else if (command.contains("done")) {
+                String[] arrOfStr = command.split(" ", 2);
+                temp = list[Integer.valueOf(arrOfStr[1])-1].replace('✗','✓');
+                list[Integer.valueOf(arrOfStr[1])-1] = temp;
+                System.out.print(border + " Nice! I've marked this task as done:\n" );
+                System.out.println("  " + list[Integer.valueOf(arrOfStr[1])-1]);
```
Comment (zhixiangteoh) [inline src/main/java/Duke.java]: Apart from extracting as method and overall improving readability, would it be better if `Integer.valueOf(arrOfStr[1]-1` is assigned to a variable before accessing it twice, on lines 38 and 41 as such? I think it would improve readability, especially if the variable is named appropriately.

**Review by zhixiangteoh — COMMENTED** [review]
It would be much better if the code was split and organized into separate methods and class files, but in general it still wasn't hard to follow because the main while loop is logically sound. Works for the first few levels but do encapsulate/abstract for the higher levels, as it would make the code more maintainable, for higher levels to come!

File: src/main/java/Duke.java [hunk 12]
```diff
@@ -1,10 +1,81 @@
...
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        String border = "____________________________________________________________\n";
+        System.out.print(border + " Hello! I'm Duke\n" + " What can I do for you?\n" + border);
+        String command;
+        int size = 0;
+        String temp;
```
Comment (manaswini05) [inline src/main/java/Duke.java]: "temp" is a vague name - consider following the format to convey the exact purpose

File: src/main/java/Duke.java [hunk 13]
```diff
@@ -1,10 +1,81 @@
...
+        String tag = "[ ]";
+        String[] temp2 = new String[2];
```
Comment (manaswini05) [inline src/main/java/Duke.java]: "temp2" is a vague name - consider following the format to convey the exact purpose

File: src/main/java/Duke.java [hunk 14]
```diff
@@ -1,10 +1,81 @@
...
+        String[] list = new String[100];
```
Comment (manaswini05) [inline src/main/java/Duke.java]: "list" is a vague name - consider following the format to convey the exact purpose

Comment (manaswini05) [inline src/main/java/Duke.java]: Any purpose for 100?

File: src/main/java/Duke.java [hunk 15]
```diff
@@ -1,10 +1,81 @@
...
+                    } else if (command.contains("deadline")) {
+                        tag = "[D]";
+                        command = command.replaceAll("deadline ", "");
+                        temp2 = command.split(" /by", 2);
+                        list[size] = tag + "[✗] " + temp2[0] + " (by:" + temp2[1] + ")";
+                    } else if (command.contains("event ")) {
+                        tag = "[E]";
+                        command = command.replaceAll("event ", "");
+                        temp2 = command.split(" /at", 2);
+                        list[size] = tag + "[✗] " + temp2[0] + " (at:" + temp2[1] + ")";
```
Comment (manaswini05) [inline src/main/java/Duke.java]: Consider avoiding deep nesting and check if you could extract them out into methods for better readability
